api/v1alpha1/containerimagebuild_types.go

```diff
@@ -92,13 +93,24 @@ type ContainerImageBuildSpec struct {
 
 // ContainerImageBuildStatus defines the observed state of ContainerImageBuild
 type ContainerImageBuildStatus struct {
+	PreviousState    BuildState   `json:"-"` // NOTE: should we persist this value?
```

Comment:
Curious how this is consumed.

Reply:
This just seemed like a good piece of information to broadcast (i.e. before/after). I still need to write the consumer bits but I thought perhaps I will need to take separate actions if move from A -> B vs A -> C. I'm probably violating YAGNI.